assets/javascripts/discourse/templates/components/chat-message.hbs

```diff
@@ -20,6 +20,7 @@
   {{on "touchend" this.handleTouchEnd passive=true}}
   {{on "mouseenter" (fn this.onHoverMessage message (hash desktopOnly=true))}}
   {{on "mouseleave" (fn this.onHoverMessage null (hash desktopOnly=true))}}
+  {{track-message-visibility}}
```

Comment:
Modifiers can be namespaced using directories (just the same as components), so we could make this `{{chat/track-message-visibility}}`